base/task.jl

```diff
@@ -240,22 +271,32 @@ end
 ## lexically-scoped waiting for multiple items
 
 function sync_end(refs)
-    c_ex = CompositeException()
+    local c_ex
+    defined = false
     for r in refs
-        try
-            wait(r)
-        catch
-            if !isa(r, Task) || (isa(r, Task) && !istaskfailed(r))
-                rethrow()
+        if isa(r, Task)
+            _wait(r)
```

Comment:
x-ref: https://github.com/JuliaLang/julia/issues/32677

If the first task is blocking on a second task and the second task is throwing an exception, we will never see the exception and `@sync` will block forever.

Reply:
:+1: but out of scope for this PR.